python/GafferImageTest/CatalogueTest.py

```diff
@@ -316,25 +343,25 @@ def testPromotion( self ) :
 
 		for image in images :
 			promotedImages.addChild( image )
-			self.assertImagesEqual( readers[0]["out"], promotedOut )
+			self.assertCatalogueImagesEqual( readers[0]["out"], promotedOut )
 
 		for i, reader in enumerate( readers ) :
 			promotedImageIndex.setValue( i )
-			self.assertImagesEqual( readers[i]["out"], promotedOut )
+			self.assertCatalogueImagesEqual( readers[i]["out"], promotedOut )
 
 		s2 = Gaffer.ScriptNode()
 		s2.execute( s.serialise() )
 
 		for i, reader in enumerate( readers ) :
 			s2["b"]["imageIndex"].setValue( i )
-			self.assertImagesEqual( readers[i]["out"], s2["b"]["out"] )
+			self.assertCatalogueImagesEqual( readers[i]["out"], s2["b"]["out"] )
 
 		s3 = Gaffer.ScriptNode()
 		s3.execute( s.serialise() )
 
 		for i, reader in enumerate( readers ) :
 			s3["b"]["imageIndex"].setValue( i )
-			self.assertImagesEqual( readers[i]["out"], s3["b"]["out"] )
+			self.assertCatalogueImagesEqual( readers[i]["out"], s3["b"]["out"] )
```

Comment:
I suspect these and others could remain as `assertImagesEqual()` if we used `enabledPlug()` rather than `valuePlug()` for managing the metadata.

Reply:
No changed actually needed with the other strategy for when the data is set. See [here](https://github.com/GafferHQ/gaffer/pull/4018/commits/4cd8d0234fbd19b6f8c425991dcb65b2c4f9ed02#diff-2de959ea3d78907f9495080b210a83440e22e8cd21b202959086f85b86e05863).